contracts/interfaces/IInterchainTokenExecutable.sol

```diff
@@ -1,14 +1,19 @@
 // SPDX-License-Identifier: MIT
 
+// can we add ^ for all versions. We're enforcing version via hardhat config
 pragma solidity ^0.8.9;
 
+// TODO: Add a base implementation that requires caller of execute to be the token service
+// All interfaces should be documented, especially since external devs look at interfaces first
 // General interface for upgradable contracts
 interface IInterchainTokenExecutable {
-    function exectuteWithInterchainToken(
+    // TODO: typo
+    function executeWithInterchainToken(
         string calldata sourceChain,
+        // TODO: If we really want to keep the interface similar, we can use string here but cast it to bytes internally
```

Comment:
I think that would be too confusing to developers. Keeping an interface the same but changing the functionality is much worse than changing both.

Reply:
yeah, not seeing much benefit too